compiler/src/iree/compiler/Codegen/Common/PolynomialApproximationPass.cpp

```diff
@@ -29,28 +29,64 @@ namespace {
 class PolynomialApproximationPass final
     : public impl::PolynomialApproximationPassBase<
           PolynomialApproximationPass> {
+
+public:
+  using impl::PolynomialApproximationPassBase<
+      PolynomialApproximationPass>::PolynomialApproximationPassBase;
+
   void runOnOperation() override {
+    using PatternFunction = std::function<decltype(populateExpandTanPattern)>;
+    std::map<std::string, PatternFunction> patternMap = {
+        {"tan", populateExpandTanPattern},
+        {"sinh", populateExpandSinhPattern},
+        {"cosh", populateExpandCoshPattern},
+        {"asinh", populateExpandAsinhPattern},
+        {"acosh", populateExpandAcoshPattern},
+        {"atanh", populateExpandAtanhPattern},
+        {"powf", populateExpandPowFPattern},
+        {"fpowi", populateExpandFPowIPattern},
+        {"erf",
+         [&](RewritePatternSet &mathPatterns) {
+           this->populateErfPattern(mathPatterns);
+         }},
+    };
+
     RewritePatternSet mathPatterns(&getContext());
-    populateExpandTanPattern(mathPatterns);
-    populateExpandSinhPattern(mathPatterns);
-    populateExpandCoshPattern(mathPatterns);
-    populateExpandAsinhPattern(mathPatterns);
-    populateExpandAcoshPattern(mathPatterns);
-    populateExpandAtanhPattern(mathPatterns);
-    populateExpandPowFPattern(mathPatterns);
-    populateExpandFPowIPattern(mathPatterns);
 
+    std::vector<StringRef> noOps = splitByComma(noApproxOps);
+    for (auto &pattern : patternMap) {
+      // Skip any ops in the "do not convert" list.
+      auto it = std::find(noOps.begin(), noOps.end(), pattern.first);
+      if (it != noOps.end()) {
+        continue;
+      }
+      pattern.second(mathPatterns);
+    }
+
+    if (failed(
+            applyPatternsGreedily(getOperation(), std::move(mathPatterns)))) {
+      return signalPassFailure();
+    }
+  }
+
+  std::vector<StringRef> splitByComma(StringRef str) {
+    std::vector<StringRef> result;
+    SmallVector<StringRef, 8> parts;
+    str.split(parts, ",");
+    for (auto part : parts) {
+      result.push_back(part.str());
+    }
+    return result;
+  }
+
+  void populateErfPattern(RewritePatternSet &mathPatterns) {
```

Comment:
I think the `ExpandExp2` and `RoundEven` patterns added here are not related to `erf`?

I also looked at the `populateMathPolynomialApproximationPatterns` . That is including a lot more patterns (including `erf` expansion patterns).

This probably needs more surgery. For now can we land something that allows you to not use polynomial expansion of just erf (the mechanism you have added to command lines is fine, just saying it might need more work to untangle what `populateMathPolynomialApproximationPatterns` does to support the full end state. Till then lets start with what we need).

Reply:
> I think the ExpandExp2 and RoundEven patterns added here are not related to erf?

Honestly, I just copied relevant code here and I thought they are some prerequisite passes to erf.

>  can we land something that allows you to not use polynomial expansion of just erf

Right now this PR already do not expand poly approx for erf. See: https://github.com/iree-org/iree/pull/19697/files#diff-1e7060ee0114d084662c3bc58670af46e4d664befc63e44f384d1f5c1b7e6b87R1099